stellarphot/core.py

```diff
@@ -305,6 +310,86 @@ def _update_passbands(self):
             mask = self['passband'] == orig_pb
             self['passband'][mask] = aavso_pb
 
+    def clean(self, remove_rows_with_mask=False, **other_restrictions):
+        """
+        Return a catalog with only the rows that meet the criteria specified.
+
+        Parameters
+        ----------
+
+        catalog : `astropy.table.Table`
+            Table of catalog information. There are no restrictions on the columns.
+
+        remove_rows_with_mask : bool, optional
+            If ``True``, remove rows in which one or more of the values is masked.
+
+        other_restrictions: dict, optional
+            Key/value pairs in which the key is the name of a column in the
+            catalog and the value is the criteria that values in that column
+            must satisfy to be kept in the cleaned catalog. The criteria must be
+            simple, beginning with a comparison operator and including a value.
+            See Examples below.
+
+        Returns
+        -------
+
+        same type as object whose method was called
+            Table with filtered data
+
+        Examples
+        --------
+
+        >>> from astropy.table import Table
+        >>> from stellarphot import BaseEnhancedTable  # Any subclasses will work too
+        >>> t = Table([[1, 2, 3], [1, 2, 3]], names=('a', 'b'), masked=True)
+        >>> bet = BaseEnhancedTable(t)
+        >>> bet['a'].mask = [True, False, False]
+        >>> bet['b'].mask = [False, False, True]
+        >>> bet.clean(remove_rows_with_mask=True)
+        <BaseEnhancedTable length=1>
+        a     b
+        int64 int64
+        ----- -----
+            2     2
+
+        >>> bet.clean(a='>2')
+        <BaseEnhancedTable length=1>
+        a     b
+        int64 int64
+        ----- -----
+            3    --
+
+        """
+        comparisons = {
+            '<': np.less,
+            '=': np.equal,
+            '>': np.greater,
+            '<=': np.less_equal,
+            '>=': np.greater_equal,
+            '!=': np.not_equal
+        }
+
+        recognized_comparison_ops = '|'.join(comparisons.keys())
+        keepers = np.ones([len(self)], dtype=bool)
+
+        if remove_rows_with_mask and self.has_masked_values:
+            for c in self.columns:
+                keepers &= ~self[c].mask
+
+        for column, restriction in other_restrictions.items():
+            criteria_re = re.compile(r'({})([-+a-zA-Z0-9]+)'.format(recognized_comparison_ops))
```

Comment:
Yes to the first question, I'm not sure about the second. Either way, I think the compile should be outside the loop since nothing in the compile regex depends on the loop variables.